iot-e2e-tests/common/src/test/java/tests/integration/com/microsoft/azure/sdk/iot/provisioning/setup/ProvisioningCommon.java

```diff
@@ -448,11 +449,25 @@ private void assertProvisionedDeviceWorks(String iothubUri, String deviceId) thr
         }
     }
 
-    protected void assertProvisionedDeviceCapabilitiesAreExpected(DeviceCapabilities expectedDeviceCapabilities, String provisionedHubConnectionString) throws IOException, IotHubException
-    {
+    protected void assertProvisionedDeviceCapabilitiesAreExpected(DeviceCapabilities expectedDeviceCapabilities, String provisionedHubConnectionString) throws IOException, IotHubException, InterruptedException {
         DeviceTwin deviceTwin = DeviceTwin.createFromConnectionString(provisionedHubConnectionString);
-        Query query = deviceTwin.queryTwin("SELECT * FROM devices WHERE deviceId = '" + testInstance.provisionedDeviceId +"'");
-        assertTrue(CorrelationDetailsLoggingAssert.buildExceptionMessageDpsIndividualOrGroup("Provisioned device " + testInstance.provisionedDeviceId + "not found in expected hub", getHostName(provisioningServiceConnectionString), testInstance.groupId, testInstance.registrationId), deviceTwin.hasNextDeviceTwin(query));
+
+        boolean deviceFoundInCorrectHub = false;
+        Query query = null;
+        long startTime = System.currentTimeMillis();
+        while (!deviceFoundInCorrectHub)
```

Comment:
Doesn't this add delay of more than 4 minutes our tests where we keep querying to see if device is provisioned?

Reply:
Only if the device never becomes queryable in this hub, which should never happen